web-animations-2/Overview.bs

```diff
@@ -2353,6 +2355,191 @@ Items sorted earlier are executed before those sorted later.
+
+<dl dfn-for="animation trigger state" dfn-type=value>
+  <dt><dfn>idle</dfn>
+  <dd>
+    The [=animation effect=] associated |animation| remains in
+    its [=animation effect/before phase=] and stays at zero [=animation/start time=].
+
+  <dt><dfn>primary</dfn>
+  <dd>
+    When switched to this value the ''primary'' behavior type defined by |type| is applied to |animation|.
+
+  <dt><dfn>inverse</dfn>
+  <dd>
+    When switched to this value the ''inverse'' behavior type defined by |type| is applied to |animation|.
+
+</dl>
+
+Issue: Do we need a formal resolution on the spec of the idle state?
+
+### Animation Trigger Active Interval ### {#trigger-active-interval}
+
+Each [=animation trigger=] defines an <dfn for="animation trigger">active interval</dfn>
+and only one such interval. This interval is the segment of the timeline’s progress
+during which the trigger’s [=animation trigger state|state=] is set to ''primary''.
+
+### Animation Trigger Ranges ### {#trigger-ranges}
+
+An [=animation trigger=] has two [=animation attachment range|ranges=],
+the <dfn>default range</dfn> and the <dfn>exit range</dfn>.
+The [=exit range=] is used to replace [=default range=] and extend
+the [=animation trigger/active interval|active interval=].
+
+Issue: Should we specify that that exit range’s edges can only be equal to or greater than the default range’s edges?
+
+Depending on its most recent [=animation trigger state|state=] |state|, the
+[=animation trigger/active interval|active interval=] is defined as follows:
+
+<dl class=switch>
+  : If |state| is ''animation trigger state/primary'',
+  ::
+    Then |interval| is set to [=exit range=].
+
+  : Otherwise,
+  ::
+    |interval| is [=default range=].
+
+</dl>
+
+### Setting the timeline of an Animation Trigger ### {#setting-trigger-timeline}
+
+  <div algorithm="setting timeline of animation trigger">
+    The procedure of <dfn>setting timeline of animation trigger</dfn> |trigger|,
+    to |new timeline| which may be <code>null</code>, is as follows:
+
+    1. Let |old timeline| be the current [=timeline=] of |trigger|, if any.
+
+    1. If |new timeline| is the same object as |old timeline|, abort this procedure.
+
+    1. Let the [=timeline=] of |trigger| be |new timeline|.
+
+    1. Run the procedure to [=updating animation trigger state=] for |trigger|.
+  </div>
+
+### Updating Animation Trigger State ### {#updating-trigger-state}
+
+<div algorithm="updating animation trigger state">
+  The procedure of <dfn>updating animation trigger state</dfn> |state|
+  of an [=animation trigger=] |trigger|,
+  with a given a flag |did trigger|, and an [=animation trigger type=] |type|,
+  is updated as follows:
+
+  1. Let |did trigger| be a boolean flag that is initially false.
+
+  1. Set |state| as follows:
+    <dl class=switch>
+      : If |trigger|’s [=local time=] is [=unresolved=],
+      ::
+        1. Set |state| to ''animation trigger state/idle''.
+        1. Set |did trigger| to false.
```

Comment:
I am still confused by this. We should not reset the state used to trigger `once` animations even if the local time is unresolved. We can get an unresolved time if the user changes the browser height for example such that the scroll timeline is no longer scrollable. This shouldn't result in re-arming a `once` animation.

Reply:
OK, I think it's fine to just remove this line. Then there's no way to reset it to `false`.